src/lib.rs

```diff
@@ -2113,7 +2113,7 @@ impl Build {
                     // This assumes qcc/q++ as compiler, which is currently the only supported compiler for QNX.
                     // See for details: https://github.com/rust-lang/cc-rs/pull/1319
                     let arg = match target.arch {
-                        "i586" => "-Vgcc_ntox86_cxx",
+                        "i586" | "x86" => "-Vgcc_ntox86_cxx",
```

Comment:
Nit: `i586` is never a `target_arch` that `rustc` will produce.

```suggestion
                        "x86" => "-Vgcc_ntox86_cxx",
```